src/main/java/seedu/apollo/command/module/AddModuleCommand.java

```diff
@@ -67,18 +69,31 @@ public static void setUpLogger() {
         }
     }
 
+    public boolean isAdded(ModuleList moduleList, Module module) {
+        if (moduleList.contains(module)) {
+            return true;
+        }
+        return false;
+    }
+
     @Override
     public void execute(TaskList taskList, Ui ui, Storage storage, ModuleList moduleList) {
         try {
-            if (module != null) {
+            if ((module != null) && (isAdded(moduleList, module) == false)) {
                 moduleList.add(module);
                 moduleList.sortModules();
                 ui.printAddModuleMessage(module);
+
+            } else if (isAdded(moduleList, module) == true) {
```

Comment:
Perhaps you could move `else if (isAdded(moduleList, module) == true)` above and throw exception, that way you don't have to check again and it simplifies the expression.